doc/manual_experimental.rst

```diff
@@ -105,23 +66,106 @@ The `void` type is only valid for parameters and return types; other symbols
 cannot have the type `void`.
 
 
-Automatic dereferencing
-=======================
+Unicode Operators
+=================
 
-Automatic dereferencing is performed for the first argument of a routine call.
-This feature has to be enabled via `{.experimental: "implicitDeref".}`:
+Under the `--experimental:unicodeOperators` switch these Unicode operators are
+also parsed as operators::
+
+  ∙ ∘ × ★ ⊗ ⊘ ⊙ ⊛ ⊠ ⊡ ∩ ∧ ⊓   # same priority as * (multiplication)
+  ± ⊕ ⊖ ⊞ ⊟ ∪ ∨ ⊔             # same priority as + (addition)
+
+
+If enabled, Unicode operators can be combined with non-Unicode operator
+symbols. The usual precedence extensions then apply, for example, `⊠=` is an
+assignment like operator just like `*=` is.
+
+No Unicode normalization step is performed.
+
+**Note**: Due to parser limitations one **cannot** enable this feature via a
+pragma `{.experimental: "unicodeOperators".}` reliably.
+
+
+Overloadable enum value names
+=============================
+
+Enabled via `{.experimental: "overloadableEnums".}`.
+
+Enum value names are overloadable much like routines. When an overloaded
+enum member is used, it produces a closed sym choice construct, here
+written as `(E|E)`. During overload resolution the right `E` is picked,
+if possible. For (array/object...) constructors the right `E` is picked,
+comparable to how `[byte(1), 2, 3]` works, one needs to use `[T.E, E2, E3]`.
+Ambiguous enum values produce a static error.
 
 .. code-block:: nim
+    :test: "nim c $1"
 
-  {.experimental: "implicitDeref".}
+  {.experimental: "overloadableEnums".}
 
-  proc depth(x: NodeObj): int = ...
+  type
+    E1 = enum
+      value1,
+      value2
+    E2 = enum
+      value1,
+      value2 = 4
 
-  var
-    n: Node
-  new(n)
-  echo n.depth
-  # no need to write n[].depth either
+  const
+    Lookuptable = [
+      E1.value1: "1",
+      value2: "2"
+    ]
+
+  proc p(e: E1) =
+    # disambiguation in 'case' statements:
+    case e
+    of value1: echo "A"
+    of value2: echo "B"
+
+  p value2
+
+
+Package level objects
+=====================
+
+Every Nim module resides in a (nimble) package. An object type can be attached
+to the package it resides in. If that is done, the type can be referenced from
+other modules as an `incomplete`:idx: object type. This feature allows to
+break up recursive type dependencies across module boundaries. Incomplete
+object types are always passed `byref` and can only be used in pointer like
+contexts (`var/ref/ptr IncompleteObject`) in general since the compiler does
+not yet know the size of the object. To complete an incomplete object
+the `package` pragma has to be used. `package` implies `byref`.
+
+As long as a type `T` is incomplete, neither `sizeof(T)` nor runtime
```

Comment:
`sizeof(T)` seems to work and returns 1 for an empty `type Pack.T = object`, is this intended?

Reply:
`sizeof(T)` works in Nim but not C. If you try to use the type the C compiler gives an error. I don't know if this is worth clarifying. I don't even really get this feature, if "packages" are what I think they mean then the "Pack" example for one should be lowercase, but because I'm not sure I just kept it the same.

Reply:
Removed mention of `sizeof`